test/components/views/dialogs/SpotlightDialog-test.tsx

```diff
@@ -449,4 +449,59 @@ describe("Spotlight Dialog", () => {
             expect(screen.getByText(potatoRoom.name!)).toBeInTheDocument();
         });
     });
+
+    describe("user directory non-matrix attributes", () => {
+        interface IUserExternalAttributesChunkMember extends IUserChunkMember {
+            location?: string;
+        }
+
+        const testPersonExternalAttributtes: IUserExternalAttributesChunkMember = {
+            user_id: "@earthling:matrix.org",
+            display_name: "Earth Person",
+            avatar_url: undefined,
+            location: "Earth", // this attribute could come from a matrix homeserver connected to LDAP
+        };
+
+        let mockedClient: MatrixClient;
+        beforeEach(() => {
+            const users: IUserExternalAttributesChunkMember[] = [testPersonExternalAttributtes]
+            mockedClient = mockClient({ rooms: [], users });
+            /* overwrite the function searchUserDirectory, to include non-matrix search attributes, used by the homeserver  */
+            mockedClient.searchUserDirectory = jest.fn(({ term, limit }) => {
+                const searchTerm = term?.toLowerCase();
+                const results = users.filter(
+                    (it) => {
+                        return (
+                            !searchTerm ||
+                            it.user_id.toLowerCase().includes(searchTerm) ||
+                            it.display_name?.toLowerCase().includes(searchTerm) ||
+                            it.location?.toLowerCase().includes(searchTerm)
+                        )
+                    }
+                );
```

Comment:
it seems superfluous to not just do
```suggestion
                const results = users;
```
while this illustrates how the server works, i feel it overcomplicates things without being useful in this place.